src/operator-run.c

```diff
@@ -399,14 +421,20 @@ static void compute_global_average_pooling_unipass(
 }
 
 static void compute_global_average_pooling_multipass(
-    const struct global_average_pooling_context context[restrict static 1],
+    const struct global_average_pooling_context context[RESTRICT_STATIC 1],
     size_t batch_index)
 {
   const void* input =
     (const void*) ((uintptr_t) context->input + batch_index * context->input_batch_stride);
   void* output =
     (void*) ((uintptr_t) context->output + batch_index * context->output_batch_stride);
-  QNNP_ALIGN(16) int32_t multipass_buffer[context->packed_channels];
+  QNNP_ALIGN(16)
+  #ifdef _MSC_VER
+    int32_t* multipass_buffer = _malloca(sizeof(int32_t) * context->packed_channels);
```

Comment:
QNNP_ALIGN(16) now applies to the pointer, not the allocated buffer. This would crash in runtime.